clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEvent.java

```diff
@@ -20,10 +20,23 @@
  * This is the abstract definition of the events created by the KafkaConsumer API
  */
 abstract public class ApplicationEvent {
+    public final Type type;
+
+    protected ApplicationEvent(Type type) {
+        this.type = type;
+    }
     /**
      * process the application event. Return true upon succesful execution,
      * false otherwise.
      * @return true if the event was successfully executed; false otherwise.
      */
-    public abstract boolean process();
+
+    @Override
+    public String toString() {
+        return type + " ApplicationEvent";
+    }
+    public enum Type {
```

Comment:
Perhaps we can leave this for a follow-up since we are not implementing the COMMIT type here.

Reply:
this is used in the stub commit application event in the prototype.